### DISCUSSION TITLE
MapDynamicControllerRoute results in "Page not found"

### DISCUSSION THREAD

**Discussion by craftyweb** [OrchardCMS/OrchardCore#14241] [Q&A] [answered]
Hello,

I'm currently building a feature to allow my clients to create more custom urls for their sitemap.

I've made a setting where the client can choose the prefix, keywords and locations.

An example could be: ~/what-do-we-do/programming/antwerp

- "what-do-we-do" is the prefix from the settings
- "programming" is an item in the keyword list
- "antwerp" is an item in the locations list.

I've tried to do it the same way as the homepage with a DynamicRouteValueTransformer:

In my startup.cs:
![image](https://github.com/OrchardCMS/OrchardCore/assets/61787638/b9557552-e80a-4be6-b3e8-a586eb2f165d)

In my KeywordLocationTransformer:
![image](https://github.com/OrchardCMS/OrchardCore/assets/61787638/0eace98a-3893-4496-a270-410c38788ad5)

My KeywordLocationController:
![image](https://github.com/OrchardCMS/OrchardCore/assets/61787638/344e88ae-7b6e-48b0-bdd0-079e2755ebff)

When I go to the url above the routevalue is created and I was expecting it would go to my custom controller. Instead it showed me "Page could not be found"

Just for making sure I did everything well I implemented the same code as the HomeRouteTransformer and my route now shows the homepage as expected.

It seems my custom controller is not reachable. What do I do wrong?

Thanks

**Comment by ns8482e** [comment]
Are you able to access your page directly using `/Crafty.OrchardCore.Foundation/KeywordLocation/Index` ?

**Reply by craftyweb** [reply, marked as answer]
Thanks for the response. Weird enough it works when I copied all of my code to a new Module. So I guess there was nothing wrong with the implementation. Don't know what happened there.
